kotest-framework/kotest-framework-plugin-gradle/src/main/kotlin/io/kotest/framework/gradle/KotestPlugin.kt

```diff
@@ -155,6 +167,33 @@ abstract class KotestPlugin : Plugin<Project> {
       }
    }
 
+   //TODO wire this to JVM test Sources automatically
+   private fun KotlinDependencyHandler.addKotestJvmRunner() {
+      project.logger.info("  Adding Kotest JUnit runner")
+      implementation("io.kotest:kotest-runner-junit5-jvm")
+   }
+
+   /**
+    * 1. Checks for the presence of KSP and stops execution if missing
+    * 2. Wires the Kotest symbol procesor for every configured KMP target
+    */
+   internal fun KotlinMultiplatformExtension.wireKotestKsp() {
+      if (!project.pluginManager.hasPlugin("com.google.devtools.ksp")) {
+         throw StopExecutionException(
+            "KSP neither found in root project nor ${project.name}, " +
+               "please add 'com.google.devtools.ksp' to the either project's plugins"
+         )
+      }
+
+      val version = System.getProperty("kotestVersion")
+      project.configurations.whenObjectAdded {
+         if (name.startsWith("ksp") && name.endsWith("Test")) {
+            project.dependencies.add(name, "io.kotest:kotest-framework-symbol-processor-jvm:$version")
+         }
+      }
+   }
```

Comment:
This prevents wiring to targets that cannot be built on the current host:

```suggestion
            val target = name.substring(3, name.length - 4).replaceFirstChar { it.lowercase() }
            if (project.getBuildableTargets().firstOrNull { target == it.name } != null)
              project.dependencies.add(name, "io.kotest:kotest-framework-symbol-processor-jvm:$version")
         }
      }
   }

fun Project.getBuildableTargets() =
    project.extensions.getByType<KotlinMultiplatformExtension>().targets.filter { target ->
        when {
            // Non-native targets are always buildable
            target.platformType != org.jetbrains.kotlin.gradle.plugin.KotlinPlatformType.native -> true
            else -> {
                runCatching {
                    val konanTarget = (target as? KotlinNativeTarget)
                    konanTarget?.publishable == true
                }.getOrElse { false }
            }
        }
    }